SI-6795 Adds negative check for "abstract override" on types in traits

JamesIry opened the pull request:
"abstract override" shouldn't was being allowed on types in traits but the result
made no sense and the spec says that shouldn't be allowed.

Commit:
SI-6795 Adds negative check for "abstract override" on types in traits

"abstract override" shouldn't was being allowed on types in traits but the result
made no sense and the spec says that shouldn't be allowed.

paulp commented on src/compiler/scala/tools/nsc/typechecker/ContextErrors.scala:
```diff
@@ -1077,7 +1077,7 @@ trait ContextErrors {
 
           case AbstractOverride =>
-            "`abstract override' modifier only allowed for members of traits"
+            "`abstract override' modifier only allowed for non-type members of traits"
```
I'm afraid you should call them "value members".

JamesIry replied:
We'll have two distinct errors: the old one for "only allowed for members of traits" and a new one for "not allowed on type members"

Commit:
SI-6795 Simplify errors related to "abstract override" on type members

Instead of saying "only allowed on non-type members of traits" use
separate errors for "not allowed on types" and "only allowed on members
of traits"

paulp commented:
lgtm.

scala-jenkins commented:
Started jenkins job pr-rangepos at https://scala-webapps.epfl.ch/jenkins/job/pr-rangepos/1133/

scala-jenkins commented:
jenkins job pr-rangepos: Success - https://scala-webapps.epfl.ch/jenkins/job/pr-rangepos/1133/

scala-jenkins commented:
Started jenkins job pr-scala-testsuite-linux-opt at https://scala-webapps.epfl.ch/jenkins/job/pr-scala-testsuite-linux-opt/1845/

scala-jenkins commented:
jenkins job pr-scala-testsuite-linux-opt: Failed - https://scala-webapps.epfl.ch/jenkins/job/pr-scala-testsuite-linux-opt/1845/

paulp commented:
"Failed" is random-build-abortese for "Succeeded."

```
docs.jline:
    [mkdir] Created dir: /localhome/jenkins/a/workspace/pr-scala-testsuite-linux-opt/build/scaladoc/jline
 [scaladoc] Documenting 34 source files to /localhome/jenkins/a/workspace/pr-scala-testsuite-linux-opt/build/scaladoc/jline
 [scaladoc] model contains 38 documentable templates
Build was aborted
Archiving artifacts
Finished: ABORTED
```

JamesIry commented:
PLS REBUILD pr-scala-testsuite-linux-opt

scala-jenkins commented:
Started jenkins job pr-scala-testsuite-linux-opt at https://scala-webapps.epfl.ch/jenkins/job/pr-scala-testsuite-linux-opt/1858/

scala-jenkins commented:
jenkins job pr-scala-testsuite-linux-opt: Success - https://scala-webapps.epfl.ch/jenkins/job/pr-scala-testsuite-linux-opt/1858/

JamesIry commented:
That was a mistake. Deleted the wrong branch.
